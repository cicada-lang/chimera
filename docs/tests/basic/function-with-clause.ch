function greet(x) {
  clause Hi("hi")

  // The expressions in `find` are implicitly `quote`d.

  let results = find q {
    Hi(p)
    Equal(q, [p, eval x])
  }

  print results
}

print greet

compute greet("xieyuheng")
