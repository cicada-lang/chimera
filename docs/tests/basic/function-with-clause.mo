function greet(x) {
  clause Hi("hi")

  // The expressions in `find` are implicitly `quote`d.

  let results = find q {
    Hi(p)
    q = [p, unquote x]
  }

  print results
}

print greet

compute greet("xieyuheng")
