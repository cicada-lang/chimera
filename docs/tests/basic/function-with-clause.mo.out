(x) => {
  Hi("hi")
  let results = find q {
    Hi(p)
    Equal(q, [p, eval x])
  }
  print results
}
[["hi", "xieyuheng"]]
